### PR TITLE
feat: Add support for GitHub Actions to automatically build a docker image on push to main branch

### PR DESCRIPTION
**Pull request by shuguet**
* Updated version to reflect `main` branch renaming.

Initial PR (now closed) here: https://github.com/HeyPuter/puter/pull/102

1) This will initiate a build of the docker image automatically, through GitHub Actions, each time a commit is pushed into the `main` branch. It requires that GitHub Actions be enabled and allowed to run on the repo.

2) The output of the action will create a package in the GitHub Container Registry (visible initially only to admins here: https://github.com/orgs/HeyPuter/packages/container/puter/settings)

3) An admin of the org will need to make the initial version of the package public, by going to that page and scrolling all the way to the bottom of the page and clicking on "Change Visibility" red button, selecting `Public`, then following the instructions.

Once this is done on the first build, subsequent builds will automatically be public.

The tagging logic is very simple for now, it will apply a tag for the sha of the image, another for the branch (so for now, only master) and another one for `latest`.
This last one can be controversial, as it means people can simply run `docker run --rm -it -p 4000:4000 ghcr.io/HeyPuter/puter`
(or the equivalent docker-compose) to get puter going, without having to pull the code/clone the repo, as this will be done on GH side automatically.

All those rules can be tweaked in the appropriate section of the `.github/workflows/docker-image.yaml` file (Lines 56 to 63 at the time of writing this).

Thank you for taking the time to read and consider this PR, I made it more for me originally, to simplify automated builds of puter from sources (get new features, bug and security fixes, work with containrrr's Watchtower, etc.), but I thought it could be useful for others :)

### CODE REVIEW

**Comment by shuguet** [comment]
@jelveh please continue here from https://github.com/HeyPuter/puter/pull/102.

You had tagged @KernelDeimos to review this (in this comment: https://github.com/HeyPuter/puter/pull/102#issuecomment-1998708373), so I'm taking the liberty to tag him again.

Thank you for renaming the branch!

**Comment by shuguet** [comment]
Also, CLA had been signed on the previous PR (closed due to branch renaming), linking that here too: https://github.com/HeyPuter/puter/pull/102#issuecomment-1998685345

File: .github/workflows/docker-image.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  push:
+    branches: ['main']
+  
+
+# Defines two custom environment variables for the workflow. These are used
+# for the Container registry domain, and a name for the Docker image that
+# this workflow builds.
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
```
Comment (KernelDeimos) [inline .github/workflows/docker-image.yaml:16]: If we want to add a suffix here, is `${{ github.repository }}-latest` the correct syntax?

Comment (shuguet) [reply .github/workflows/docker-image.yaml:16]: That will change the name of the image/package created yes.
The result would be that the docker command would look like this instead:
`docker run --rm -it -p 4000:4000 ghcr.io/puter/puter-latest`

If you're looking at changing the `tag` instead (the part that comes at the end of the docker image, after a semi-colon `: `) then you need to change this section instead:
https://github.com/HeyPuter/puter/pull/104/files#diff-a9b4f1a51dd81dfaef1e9d563dfe6045ab8089a05200453d6e0b2d66201f419fR55-R62

**Comment by jelveh** [comment]
@shuguet Thank you very much for your contribution!

**Comment by shuguet** [comment]
> @shuguet Thank you very much for your contribution!

You're welcome :)

Also, @jelveh, can you also follow step 3 of the instruction in the first message of this PR?
This:
> An admin of the org will need to make the initial version of the package public, by going to that page and scrolling all the way to the bottom of the page and clicking on "Change Visibility" red button, selecting Public, then following the instructions.

Otherwise, the package/container image will not be visible :(

**Comment by jelveh** [comment]
> The good news is, it works!

It sure does! thanks to you :)

I made the package public. We should be all set now.

**Comment by shuguet** [comment]
Can confirm, we're all set!:
```
$ docker run --rm -p 4000:4000 ghcr.io/heyputer/puter

> puter.com@1.0.0 start
> nodemon --exec "node dev-server.js"

[nodemon] 2.0.22
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): *.*
[nodemon] watching extensions: js,json,mjs,jsx,svg,css
[nodemon] starting `node dev-server.js`

-----------------------------------------------------------

Puter is now live at:  http://localhost:4000

-----------------------------------------------------------
```

**Pull request merged**